### ISSUE TITLE
Continued erros

### ISSUE THREAD

**Issue by midd33** [FlyingDiver/Indigo-Ecobee-2#25]
I keep getting this error. Only have one thermostat.  I tried downgrading to 2.4.0 with no luck

```
Jul 19, 2021 at 9:49:08 PM
   Ecobee 2 Error                  Error in plugin execution runConcurrentThread:

Traceback (most recent call last):
  File "plugin.py", line 210, in runConcurrentThread
  File "/Library/Application Support/Perceptive Automation/Indigo 7.5/Plugins/Ecobee 2.indigoPlugin/Contents/Server Plugin/ecobee_devices.py", line 369, in update
  File "/Library/Application Support/Perceptive Automation/Indigo 7.5/Plugins/Ecobee 2.indigoPlugin/Contents/Server Plugin/temperature_scale.py", line 20, in convertFromEcobee
ValueError: could not convert string to float: unknown

   Ecobee 2 Error                  plugin runConcurrentThread function returned or failed (will attempt again in 10 seconds)

Jul 19, 2021 at 9:53:39 PM
   Z-Wave                          received "Front Door" status update locked (manually)

Jul 19, 2021 at 9:54:21 PM
   Ecobee 2 Debug                  logLevel = 5
   Ecobee 2 Debug                  updateFrequency = 900.0
   Ecobee 2 Debug                  setting temperature scale to F
   Ecobee 2 Debug                  Ecobee Account (422790907): Ecobee Account Update OK, got info on 1 thermostats
   Ecobee 2 Debug                  [
    {
        "brand": "ecobee",
        "equipmentStatus": "",
        "events": [
            {
                "coolHoldTemp": 700,
                "coolRelativeTemp": 0,
                "drRampUpTemp": 0,
                "drRampUpTime": 3600,
                "dutyCyclePercentage": 255,
                "endDate": "2035-12-29",
                "endTime": "12:00:00",
                "fan": "auto",
                "fanMinOnTime": 5,
                "fanSpeed": "optimized",
                "heatHoldTemp": 700,
                "heatRelativeTemp": 0,
                "holdClimateRef": "",
                "isCoolOff": false,
                "isHeatOff": false,
                "isOccupied": true,
                "isOptional": true,
                "isTemperatureAbsolute": true,
                "isTemperatureRelative": false,
                "linkRef": "",
                "name": "hold",
                "occupiedSensorActive": false,
                "running": true,
                "startDate": "2021-07-19",
                "startTime": "09:21:05",
                "type": "hold",
                "unoccupiedSensorActive": false,
                "vent": "off",
                "ventilatorMinOnTime": 20
            }
        ],
        "features": "Home,HomeKit",
        "identifier": "311069263249",
        "isRegistered": true,
        "lastModified": "2021-07-19 13:22:04",
        "modelNumber": "athenaSmart",
        "name": "Home",
        "program": {
            "climates": [
                {
                    "climateRef": "home",
                    "colour": 13560055,
                    "coolFan": "auto",
                    "coolTemp": 740,
                    "heatFan": "auto",
                    "heatTemp": 680,
                    "isOccupied": true,
                    "isOptimized": false,
                    "name": "Home",
                    "owner": "system",
                    "sensors": [
                        {
                            "id": "ei:0:1",
                            "name": "Home"
                        },
                        {
                            "id": "rs:100:1",
                            "name": "Bedroom"
                        },
                        {
                            "id": "rs:101:1",
                            "name": "Sydney's"
                        }
                    ],
                    "type": "program",
                    "vent": "off",
                    "ventilatorMinOnTime": 20
                },
                {
                    "climateRef": "sleep",
                    "colour": 2179683,
                    "coolFan": "auto",
                    "coolTemp": 730,
                    "heatFan": "auto",
                    "heatTemp": 630,
                    "isOccupied": true,
                    "isOptimized": false,
                    "name": "Sleep",
                    "owner": "system",
                    "sensors": [
                        {
                            "id": "ei:0:1",
                            "name": "Home"
                        },
                        {
                            "id": "rs:100:1",
                            "name": "Bedroom"
                        },
                        {
                            "id": "rs:101:1",
                            "name": "Sydney's"
                        }
                    ],
                    "type": "program",
                    "vent": "off",
                    "ventilatorMinOnTime": 20
                },
                {
                    "climateRef": "away",
                    "colour": 9021815,
                    "coolFan": "auto",
                    "coolTemp": 740,
                    "heatFan": "auto",
                    "heatTemp": 640,
                    "isOccupied": false,
                    "isOptimized": true,
                    "name": "Away",
                    "owner": "system",
                    "sensors": [
                        {
                            "id": "ei:0:1",
                            "name": "Home"
                        },
                        {
                            "id": "rs:100:1",
                            "name": "Bedroom"
                        },
                        {
                            "id": "rs:101:1",
                            "name": "Sydney's"
                        }
                    ],
                    "type": "program",
                    "vent": "off",
                    "ventilatorMinOnTime": 20
                }
            ],
            "currentClimateRef": "home",
            "schedule": [
                [
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep"
                ],
                [
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep"
                ],
                [
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep"
                ],
                [
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "away",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep"
                ],
                [
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep"
                ],
                [
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep"
                ],
                [
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "home",
                    "sleep",
                    "sleep",
                    "sleep",
                    "sleep"
                ]
            ]
        },
        "remoteSensors": [
            {
                "capability": [
                    {
                        "id": "1",
                        "type": "temperature",
                        "value": "687"
                    },
                    {
                        "id": "2",
                        "type": "occupancy",
                        "value": "false"
                    }
                ],
                "code": "D3KN",
                "id": "rs:100",
                "inUse": true,
                "name": "Bedroom",
                "type": "ecobee3_remote_sensor"
            },
            {
                "capability": [
                    {
                        "id": "1",
                        "type": "temperature",
                        "value": "unknown"
                    },
                    {
                        "id": "2",
                        "type": "occupancy",
                        "value": "false"
                    }
                ],
                "code": "HZCM",
                "id": "rs:101",
                "inUse": false,
                "name": "Sydney's",
                "type": "ecobee3_remote_sensor"
            },
            {
                "capability": [
                    {
                        "id": "1",
                        "type": "temperature",
                        "value": "713"
                    },
                    {
                        "id": "2",
                        "type": "humidity",
                        "value": "49"
                    },
                    {
                        "id": "3",
                        "type": "occupancy",
                        "value": "true"
                    }
                ],
                "id": "ei:0",
                "inUse": true,
                "name": "Home",
                "type": "thermostat"
            }
        ],
        "runtime": {
            "actualAQAccuracy": 0,
            "actualAQScore": -5002,
            "actualCO2": -5002,
            "actualHumidity": 49,
            "actualTemperature": 700,
            "actualVOC": -5002,
            "connectDateTime": "2021-07-18 11:43:42",
            "connected": true,
            "desiredCool": 700,
            "desiredCoolRange": [
                650,
                920
            ],
            "desiredDehumidity": 60,
            "desiredFanMode": "auto",
            "desiredHeat": 700,
            "desiredHeatRange": [
                574,
                744
            ],
            "desiredHumidity": 36,
            "disconnectDateTime": "2021-07-15 03:17:38",
            "firstConnected": "2017-12-29 02:27:01",
            "lastModified": "2021-07-20 01:53:58",
            "lastStatusModified": "2021-07-20 01:53:58",
            "rawTemperature": 700,
            "runtimeDate": "2021-07-20",
            "runtimeInterval": 19,
            "runtimeRev": "210720015358",
            "showIconMode": 0
        },
        "settings": {
            "autoAway": true,
            "autoHeatCoolFeatureEnabled": true,
            "auxMaxOutdoorTemp": 700,
            "auxOutdoorTempAlert": 500,
            "auxOutdoorTempAlertNotify": false,
            "auxOutdoorTempAlertNotifyTechnician": false,
            "auxRuntimeAlert": 10800,
            "auxRuntimeAlertNotify": false,
            "auxRuntimeAlertNotifyTechnician": false,
            "backlightOffDuringSleep": false,
            "backlightOffTime": 60,
            "backlightOnIntensity": 10,
            "backlightSleepIntensity": 4,
            "coldTempAlert": 500,
            "coldTempAlertEnabled": true,
            "compressorProtectionMinTemp": 350,
            "compressorProtectionMinTime": 300,
            "condensationAvoid": false,
            "coolMaxTemp": 1200,
            "coolMinTemp": -100,
            "coolRangeHigh": 920,
            "coolRangeLow": 650,
            "coolStages": 1,
            "coolingLockout": false,
            "dehumidifierLevel": 60,
            "dehumidifierMode": "off",
            "dehumidifyOvercoolOffset": 0,
            "dehumidifyWhenHeating": false,
            "dehumidifyWithAC": false,
            "disableAlertsOnIdt": false,
            "disableHeatPumpAlerts": false,
            "disablePreCooling": false,
            "disablePreHeating": false,
            "drAccept": "always",
            "eiLocation": "",
            "electricityBillCycleMonths": 1,
            "electricityBillStartMonth": 1,
            "electricityBillingDayOfMonth": 1,
            "enableElectricityBillAlert": false,
            "enableProjectedElectricityBillAlert": false,
            "fanControlRequired": true,
            "fanMinOnTime": 5,
            "fanSpeed": "low",
            "followMeComfort": false,
            "groupName": "",
            "groupRef": "",
            "groupSetting": 0,
            "hasBoiler": false,
            "hasDehumidifier": false,
            "hasElectric": false,
            "hasErv": false,
            "hasForcedAir": true,
            "hasHeatPump": false,
            "hasHrv": false,
            "hasHumidifier": true,
            "hasUVFilter": true,
            "heatCoolMinDelta": 50,
            "heatMaxTemp": 1200,
            "heatMinTemp": 450,
            "heatPumpGroundWater": false,
            "heatPumpReversalOnCool": true,
            "heatRangeHigh": 744,
            "heatRangeLow": 574,
            "heatStages": 1,
            "holdAction": "indefinite",
            "hotTempAlert": 920,
            "hotTempAlertEnabled": true,
            "humidifierMode": "manual",
            "humidity": "36",
            "humidityAlertNotify": false,
            "humidityAlertNotifyTechnician": false,
            "humidityHighAlert": -1,
            "humidityLowAlert": -1,
            "hvacMode": "cool",
            "installerCodeRequired": false,
            "isRentalProperty": false,
            "isVentilatorTimerOn": false,
            "lastServiceDate": "2019-11-07",
            "locale": "en",
            "maxSetBack": 100,
            "maxSetForward": 80,
            "monthlyElectricityBillLimit": 0,
            "monthsBetweenService": 6,
            "quickSaveSetBack": 40,
            "quickSaveSetForward": 40,
            "randomStartDelayCool": 0,
            "randomStartDelayHeat": 0,
            "remindMeDate": "2021-05-07",
            "serviceRemindMe": true,
            "serviceRemindTechnician": false,
            "smartCirculation": false,
            "soundAlertVolume": 0,
            "soundTickVolume": 0,
            "stage1CoolingDifferentialTemp": 5,
            "stage1CoolingDissipationTime": 31,
            "stage1HeatingDifferentialTemp": 5,
            "stage1HeatingDissipationTime": 31,
            "tempAlertNotify": true,
            "tempAlertNotifyTechnician": false,
            "tempCorrection": 0,
            "useCelsius": false,
            "useTimeFormat12": true,
            "useZoneController": false,
            "userAccessCode": "",
            "userAccessSetting": 0,
            "vent": "off",
            "ventilatorDehumidify": true,
            "ventilatorFreeCooling": true,
            "ventilatorMinOnTime": 20,
            "ventilatorMinOnTimeAway": 0,
            "ventilatorMinOnTimeHome": 20,
            "ventilatorOffDateTime": "",
            "ventilatorType": "none",
            "wifiOfflineAlert": false
        },
        "thermostatRev": "210719132204",
        "thermostatTime": "2021-07-19 21:54:22",
        "utcTime": "2021-07-20 01:54:22"
    }
]
   Ecobee 2 Debug                  Ecobee Account (422790907): getting data for 'Home', 311069263249
   Ecobee 2 Debug                  Ecobee Account (422790907): getting data for remote sensor 'Bedroom', D3KN
   Ecobee 2 Debug                  Ecobee Account (422790907): getting data for remote sensor 'Sydney's', HZCM
   Ecobee 2 Debug                  Thermostat Update, thermostats =
{
    "311069263249": {
        "actualHumidity": 49,
        "actualTemperature": 700,
        "brand": "ecobee",
        "climates": {
            "away": "Away",
            "home": "Home",
            "sleep": "Sleep"
        },
        "currentClimate": "home",
        "desiredCool": 700,
        "desiredFanMode": "auto",
        "desiredHeat": 700,
        "equipmentStatus": "",
        "fanMinOnTime": 5,
        "features": "Home,HomeKit",
        "hvacMode": "cool",
        "internal": {
            "humidity": "49",
            "occupancy": "true",
            "temperature": "713"
        },
        "latestEventType": "hold",
        "modelNumber": "athenaSmart",
        "name": "Home",
        "remotes": {
            "D3KN": {
                "name": "Bedroom",
                "occupancy": "false",
                "temperature": "687",
                "thermostat": "311069263249"
            },
            "HZCM": {
                "name": "Sydney's",
                "occupancy": "false",
                "temperature": "unknown",
                "thermostat": "311069263249"
            }
        }
    }
}
sensors = {
    "D3KN": {
        "name": "Bedroom",
        "occupancy": "false",
        "temperature": "687",
        "thermostat": "311069263249"
    },
    "HZCM": {
        "name": "Sydney's",
        "occupancy": "false",
        "temperature": "unknown",
        "thermostat": "311069263249"
    }
}

   Ecobee 2 Debug                  Ecobee: Updating device
   Ecobee 2 Debug                  Ecobee: Reported hsp: 700, converted hsp: 70.0
   Ecobee 2 Debug                  Ecobee: Reported csp: 700, converted csp: 70.0
   Ecobee 2 Debug                  Ecobee: Reported dispTemp: 700, converted dispTemp: 70.0
   Ecobee 2 Debug                  Ecobee: Reported temp2: 713, converted temp2: 71.3
   Ecobee 2 Debug                  Ecobee Home (311069263249) Remote - Sydney's (HZCM): Updating device
   Ecobee 2 Error                  Error in plugin execution runConcurrentThread:

Traceback (most recent call last):
  File "plugin.py", line 210, in runConcurrentThread
  File "/Library/Application Support/Perceptive Automation/Indigo 7.5/Plugins/Ecobee 2.indigoPlugin/Contents/Server Plugin/ecobee_devices.py", line 369, in update
  File "/Library/Application Support/Perceptive Automation/Indigo 7.5/Plugins/Ecobee 2.indigoPlugin/Contents/Server Plugin/temperature_scale.py", line 20, in convertFromEcobee
ValueError: could not convert string to float: unknown

   Ecobee 2 Error                  plugin runConcurrentThread function returned or failed (will attempt again in 10 seconds)
   Ecobee 2 Debug                  runConcurrentThread starting```

**Comment by FlyingDiver** [comment]
I've never seen a remote sensor report the temperature as "unknown"
```    "HZCM": {
        "name": "Sydney's",
        "occupancy": "false",
        "temperature": "unknown",
        "thermostat": "311069263249"
    }
```

**Comment by FlyingDiver** [comment]
Fixed in 2.4.2
